pywt/_dwt.py

```diff
@@ -401,3 +399,115 @@ def upcoef(part, coeffs, wavelet, level=1, take=0):
     if part not in 'ad':
         raise ValueError("Argument 1 must be 'a' or 'd', not '%s'." % part)
     return np.asarray(_upcoef(part == 'a', coeffs, wavelet, level, take))
+
+
+def pad(x, pad_widths, mode):
+    """Extend a 1D signal using a given boundary mode.
+
+    This function operates like ``numpy.pad`` but supports all signal extension
+    modes that can be used by PyWavelets discrete wavelet transforms.
+
+    Parameters
+    ----------
+    x : ndarray
+        The array to pad
+    pad_widths : {sequence, array_like, int}
+        Number of values padded to the edges of each axis.
+        ((before_1, after_1), … (before_N, after_N)) unique pad widths for each
+        axis. ((before, after),) yields same before and after pad for each
+        axis. (pad,) or int is a shortcut for before = after = pad width for
+        all axes.
+    mode : str, optional
+        Signal extension mode, see :ref:`Modes <ref-modes>`.
+
+    Returns
+    -------
+    pad : ndarray
+        Padded array of rank equal to array with shape increased according to
+        `pad_width`.
+
+    Notes
+    -----
+    The performance of padding in dimensions > 1 may be substantially slower
+    for modes ``smooth`` and ``antisymmetric`` as these modes are not supported
+    efficiently by the underlying ``numpy.pad`` function.
+    """
+    x = np.asanyarray(x)
+
+    # process pad_widths exactly as in numpy.pad
+    pad_widths = np.array(pad_widths)
+    pad_widths = np.round(pad_widths).astype(np.intp, copy=False)
+    if pad_widths.min() < 0:
+        raise ValueError("pad_widths must be > 0")
+    pad_widths = np.broadcast_to(pad_widths, (x.ndim, 2)).tolist()
+
+    if mode in ['symmetric', 'reflect']:
+        xp = np.pad(x, pad_widths, mode=mode)
+    elif mode in ['periodic', 'periodization']:
+        if mode == 'periodization':
+            # Promote odd-sized dimensions to even length by duplicating the
+            # last value.
+            edge_pad_widths = [(0, x.shape[ax] % 2)
+                               for ax in range(x.ndim)]
+            x = np.pad(x, edge_pad_widths, mode='edge')
+        xp = np.pad(x, pad_widths, mode='wrap')
+    elif mode == 'zero':
+        xp = np.pad(x, pad_widths, mode='constant', constant_values=0)
+    elif mode == 'constant':
```

Comment:
perhaps makes sense to point out in the `Notes` section of the docstring that the meaning of `'constant'` differs between this function and `np.pad`?